lib/facility.rb

```diff
@@ -1,14 +1,49 @@
+require "./lib/vehicle"
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_hash)
+    @name = facility_hash[:name]
+    @address = facility_hash[:address]
+    @phone = facility_hash[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    @registered_vehicles << vehicle
+    @collected_fees += registration_fee(vehicle)
+    @registration_date = Date.today
+
```

Comment:
This method is on the right track. I like that you created the `registration_fee` helper method to move out some of the logic.
One thing we needed to do before running lines 19-21 though, is make sure that the facility.services included the service of "Vehicle Registration", so before line 19 we'd want something like `if @services.include?("Vehicle Registration"` then run the logic.
We also want to think about where we want the registration date to be set. It's the vehicle object that has the attribute of registration date, so line 21 we could write like `vehicle.registration_date = Date.today`